dask_ml/model_selection/_incremental.py

```diff
@@ -347,3 +354,228 @@ def fit(
         scorer=scorer,
         random_state=random_state,
     )
+
+
+# ----------------------------------------------------------------------------
+# Base class for scikit-learn compatible estimators using fit
+# ----------------------------------------------------------------------------
+
+
+class BaseIncrementalSearchCV(DaskBaseSearchCV):
+    """Base class for estimators using the incremental `fit`.
+
+    Subclasses must implement the following abstract methods
+
+    * _get_ids_and_args : Build the dict of
+       ``{model_id : (param_list, additional_calls)}``
+        that are eventually passed to `fit`
+    * _get_cv_results : Build the dict containing the CV results.
+    * _get_best_estimator : Select the best estimator from the set of
+      fitted estimators
+    * _update_results : Update self with attributes from the fitted
+      esimators.
```

Comment:
I would also expect this to be somewhat standardized among different implementations.  What does this need to do?

Reply:
I'm not really sure. For Hyperband, it's something like

```python

        self.history_ = sum([SHA._history for SHA in SHAs.values()], [])
        self.metadata_ = {
            "models": sum(m["models"] for m in meta),
            "partial_fit_calls": sum(m["partial_fit_calls"] for m in meta),
            "brackets": meta,
        }

```

This could probably have a default implementation that extracts the results that will likely be useful to all base classes (I'm just not sure what those items are at the moment).

Reply:
I think that my objective here is for users not to have to write too much to make things work.  Some of these hooks should probably be optional.